cmd/image-builder-db-test/main_test.go

```diff
@@ -342,19 +348,52 @@ func testClones(t *testing.T) {
 	require.Equal(t, clones[1], *entry)
 }
 
-func TestMain(t *testing.T) {
+func testBlueprints(t *testing.T) {
+	d, err := db.InitDBConnectionPool(connStr(t))
+	require.NoError(t, err)
+	conn := connect(t)
+	defer conn.Close(context.Background())
+
+	b1 := v1.BlueprintV1{
+		Version:     1,
+		Name:        "name",
+		Description: "desc",
+	}
+	body, err := json.Marshal(b1)
+	require.NoError(t, err)
+
+	id := uuid.New()
+	err = d.InsertBlueprint(id, ORGID1, ANR1, "name", "desc", 1, body)
+	require.NoError(t, err)
+
+	entry, err := d.GetBlueprint(id, ORGID1, ANR1)
+	require.NoError(t, err)
+	b2, err := v1.BlueprintFromEntry(entry)
+	require.NoError(t, err)
+	require.Equal(t, b1, b2)
+
+	err = d.DeleteBlueprint(id, ORGID1, ANR1)
+	require.NoError(t, err)
+}
+
+func runTest(t *testing.T, f func(*testing.T)) {
+	migrateTern(t)
+	defer tearDown(t)
+	f(t)
+}
+
+func TestAll(t *testing.T) {
 	fns := []func(*testing.T){
 		testInsertCompose,
 		testGetCompose,
 		testCountComposesSince,
 		testGetComposeImageType,
 		testDeleteCompose,
 		testClones,
+		testBlueprints,
 	}
 
 	for _, f := range fns {
-		migrateTern(t)
-		f(t)
-		tearDown(t)
+		runTest(t, f)
 	}
 }
```

Comment:
Renamed because TestMain is reserved for test suites, it does work but Goland is marking this as an error and it is pesky. Also refactored it to function so defer can be used to ensure all data is deleted after even an error.